shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1018,16 +1020,49 @@ module MakeImpl<LocationSig Location, InputSig<Location> Lang> {
 
     private predicate sinkNode = Stage1::sinkNode/2;
 
+    private predicate sourceLabel(NodeEx node, string label) {
+      sourceNode(node, _) and
+      exists(Node n | n = node.asNode() |
+        exists(string model | knownSourceModel(n, model) and label = model)
+        or
+        not knownSourceModel(n, _) and label = ""
+      )
+    }
+
+    private predicate sinkLabel(NodeEx node, string label) {
+      sinkNode(node, _) and
+      exists(Node n | n = node.asNode() |
+        exists(string model | knownSinkModel(n, model) and label = model)
+        or
+        not knownSinkModel(n, _) and label = ""
+      )
+    }
+
+    bindingset[label1, label2]
+    pragma[inline_late]
+    private string mergeLabels(string label1, string label2) {
```

Comment:
There's merging going on for different reasons, and it's not completely satisfactory to simply discard any 2nd+ label/model, so that's why I wanted to keep both instances private.